core/src/main/scala/org/apache/spark/serializer/JavaSerializer.scala

```diff
@@ -92,9 +114,18 @@ private object JavaDeserializationStream {
 }
 
 private[spark] class JavaSerializerInstance(
```

Comment:
Can we simply reuse the `ByteBufferOutputStream` in the `serialize` method? e.g.
```
private val bos = new ByteBufferOutputStream()
override def serialize[T: ClassTag](t: T): ByteBuffer = {
  bos.reset()
  val out = serializeStream(bos)
  out.writeObject(t)
  out.close()
  bos.toByteBuffer
}
```

Reply:
I think it's OK if we use it in a single thread. However, this is not helpful for the deserialization performance because the overhead comes from `ObjectInputStream`.